diff --git a/bin/index.js b/bin/index.js
--- a/bin/index.js
+++ b/bin/index.js
@@ -1,1 +1,7 @@
 #!/usr/bin/env node
+const path = require('path');
+const PATH = Object.entries(process.env).concat([['path', '']]).find(([key]) => /^PATH$/i.test(key))[1];
+if (
+  process.env.npm_config_prefix
+  && !PATH.split(path.delimiter).some(pathdir => path.resolve(pathdir) === path.resolve(process.env.npm_config_prefix, 'bin'))
+) delete process.env.npm_config_prefix;
